spock-core/core.gradle

```diff
@@ -51,15 +56,16 @@ jar {
   bnd(
     'Export-Package': ['org.spockframework.*', 'spock.*'].join(','),
     'Import-Package': [
-      /groovy.*;version="[${groovyVersion},${org.gradle.util.VersionNumber.parse(groovyVersion).major + 1})"/,
-      /org.codehaus.groovy.*;version="[${groovyVersion},${org.gradle.util.VersionNumber.parse(groovyVersion).major + 1})"/,
-      /org.junit.*;version="[${libs.junitBom.version},${org.gradle.util.VersionNumber.parse(libs.junitBom.version).major + 1})"/,
-      /org.hamcrest.*;version="[${libs.hamcrest.version},${org.gradle.util.VersionNumber.parse(libs.hamcrest.version).major + 1})";resolution:=optional/,
-      /org.objenesis.*;version="[${libs.objenesis.version},${org.gradle.util.VersionNumber.parse(libs.objenesis.version).major + 1})";resolution:=optional/,
-      /net.bytebuddy.*;version="[${libs.bytebuddy.version},${org.gradle.util.VersionNumber.parse(libs.bytebuddy.version).major + 1})";resolution:=optional/,
-      /net.sf.cglib.*;version="[${libs.cglib.version},${org.gradle.util.VersionNumber.parse(libs.cglib.version).major + 1})";resolution:=optional/,
-      /org.objectweb.asm.*;version="[${libs.asm.version},${org.gradle.util.VersionNumber.parse(libs.asm.version).major + 1})";resolution:=optional/,
-      'org.w3c.dom.*'].join(',')
+      /org.hamcrest.*;resolution:=optional/,
+      /org.objenesis.*;resolution:=optional/,
+      /net.bytebuddy.*;resolution:=optional/,
+      /net.sf.cglib.*;resolution:=optional/,
+      /org.objectweb.asm.*;resolution:=optional/,
```

Comment:
You removed non optional dependencies. Does it mean that the plugin adds it automatically?

Reply:
By default Bnd adds packages imports automatically :) In fact it's better to let bnd find packages that are imported than try to manage that list manually. Now, there is the case for adding non-default attributes like you have for `resolution:=optional`. So, you should start by not adding the `Import-Package` header at all; adding it only when you wish to specify things like `resolution:=optional`. However, when you do that you have to terminate the instruction with `*` to make sure bnd continues to add the rest of what it finds.